kubetools/kubernetes/api.py

```diff
@@ -41,11 +42,40 @@ def _get_k8s_apps_api(env):
     return client.AppsV1Api(api_client=api_client)
 
 
-def _get_k8s_batch_api(env):
+def check_if_cronjob_compatible(env):
+    settings = get_settings()
+
+    api_core = _get_k8s_core_api(env)
+    list_node = api_core.list_node().items
+    k8s_version = "1.21.0"
+    lowest_version = "1.21.0"
+
+    # Get the lowest kubelet version of all the nodes of the Cluster
+    for node in list_node:
+        kubelet_version = node.status.node_info.kubelet_version.replace('v', '')
+        if version.parse(kubelet_version) < version.parse(lowest_version):
+            lowest_version = kubelet_version
+
+    if version.parse(lowest_version) >= version.parse(k8s_version):
```

Comment:
```suggestion
    k8s_version = version.parse("1.21.0")
    
    nodes_versions = [
        node.status.node_info.kubelet_version.replace('v', '')
        for node in list_node
    ]
    parsed_versions = map(version.parse, nodes_versions)
    lowest_version = min(parsed_version)
    
    if lowest_version >= k8s_version:
```

* pythonic
* readable: no need for a comment, variable and function names explain what's happening
* readable: reduction in function calls and operations
* more performant: parsing of each version is done exactly once
* safer: if there is no node, this will raise a `ValueError` in the `min` call.